### PR TITLE
Initialize Go modules for pulumi/pulumi

### PR DESCRIPTION
**Pull request by jen20**
This pull request is an _experiment_ with whether or not we can now easily adopt Go modules for dependency management without extensive changes to build infrastructure.

### CODE REVIEW

**Comment by jen20** [comment]
I've marked this as "Ready for Review" to get Travis CI to run on it as well as AppVeyor. Please _do not merge_ at this point.

File: pkg/backend/display/jsonmessage.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func (jm *Progress) Display(out io.Writer, termInfo termInfo) {
 	if termInfo != nil && /*jm.Stream == "" &&*/ jm.Action != "" {
 		clearLine(out, termInfo)
 		endl = "\r"
-		fmt.Fprintf(out, endl)
+		fmt.Print(out, endl)
```
Comment (ellismg) [inline pkg/backend/display/jsonmessage.go]: I don't think this change is correct.

File: pkg/backend/display/progress.go [hunk 2]
```diff
@@ -1224,7 +1225,7 @@ func (display *ProgressDisplay) getStepInProgressDescription(step engine.StepEve
 	return op.Color() + getDescription() + colors.Reset
 }
 
-func writeString(b *bytes.Buffer, s string) {
+func writeString(b io.StringWriter, s string) {
```
Comment (ellismg) [inline pkg/backend/display/progress.go]: Just guessing this is due to the newer compiler helpfully telling us to make this change?

File: pkg/backend/display/jsonmessage.go (hunk 1 above)
Comment (jen20) [reply pkg/backend/display/jsonmessage.go]: You are correct, I’ll fix up.

File: pkg/backend/display/progress.go (hunk 2 above)
Comment (jen20) [reply pkg/backend/display/progress.go]: I think it’s the linter upgrade possibly combined with the compiler upgrade.

**Comment by jen20** [comment]
I've rebased this, updated the modules as per the previous head of `master`, and fixed up the incorrect `Print -> Fprint` change. Pending CI, I think this is good to go.

**Comment by jen20** [comment]
Thanks for looking over this @ellismg! The changes you requested are made and things are green now, so I'm going to go ahead and merge this, and start filtering the changes downstream.

**Pull request merged**